src/Components/Admins/Form/index.js

```diff
@@ -20,230 +15,212 @@ const Form = ({
     dni: '',
     password: ''
   });
+  const [repeatPass, setRepeatPass] = useState('');
+  const [toastErroOpen, setToastErroOpen] = useState(false);
+  const [toastMessage, setToastMessage] = useState('Error in database');
+  const [modalSuccessOpen, setModalSuccessOpen] = useState(false);
 
-  const [modalConfirmOpen, setModalConfirmOpen] = useState(false);
+  const history = useHistory();
+  const { id } = useParams();
+  const location = useLocation();
+  const data = location.state.params;
+
+  useEffect(() => {
+    if (data.mode === 'create') {
+      setInputValue({
+        firstName: '',
+        lastName: '',
+        phone: '',
+        email: '',
+        city: '',
+        dni: '',
+        password: ''
+      });
+    } else {
+      setInputValue({
+        firstName: data.item.firstName,
+        lastName: data.item.lastName,
+        phone: data.item.phone,
+        email: data.item.email,
+        city: data.item.city,
+        dni: data.item.dni,
+        password: data.item.password
+      });
+      setRepeatPass(data.item.password);
+      setEditMode(true);
+    }
+  }, []);
 
   const addAdmins = async () => {
     try {
-      await fetch(`${process.env.REACT_APP_API_URL}/api/admins`, {
+      const resp = await fetch(`${process.env.REACT_APP_API_URL}/api/admins`, {
         method: 'POST',
         headers: {
           'Content-Type': 'application/json'
         },
-        body: JSON.stringify(admin)
+        body: JSON.stringify(inputValue)
       });
+      const data = await resp.json();
+      if (!data.error) {
+        confirmation();
+      } else {
```

Comment:
instead of throwing new Error you could show a toast with the error message from the server, cause it may contain instructions for a correct fill of some input or something usefull for the user